### ISSUE TITLE
Feature request: Support configuration for (reply) serializer/fast-json-stringify

### ISSUE THREAD

**Issue by Retro64** [fastify/fastify#3007]
## 🚀 Feature Proposal

Extend configuration in order to pass configuration values for `fast-json-stringify` for serialisation.

## Motivation

When adding a schema for reply to my routes I used the `nullable` property. When writign tests, I was puzzled, not-nullable values were serialized without an error. I checked the `Validation and Serialisation` and follows the link to the `fast-json-stringify` documentation, showing a property to enable nullable-checks. Unfortunately it is false by default and fastify gives no opportunity to pass the value to the serializer (at least I did not find any). I am aware every additional check slows down the serialization, but at least in testing it might make sense to enable checks in the serializer.

## Example

current state:

```
  const server = fastify();
  server.get(
    '/foo',
      {
        schema: {
            $id: 'foo',
            type: 'object',
            properties: {
                bar: {
                    type: 'string',
                    nullable: false,
                }
            }
         }
      }
      ...
  );

```
Will not fail, if a `{}` would be replied

**Comment by mcollina** [comment]
cc @Eomm @jsumners I think we made this customizable recently, or maybe I was mistaken?

**Comment by Retro64** [comment]
Wow - that was a fast reply (the main reason to write this comment). At least I did not find any opportunity in the main branch were the options might be assigned

https://github.com/fastify/fastify/blob/7efd2540f1ca169f128c1dd4318512347583b293/lib/reply.js#L6
https://github.com/fastify/fastify/blob/7efd2540f1ca169f128c1dd4318512347583b293/lib/reply.js#L29

nor in the documentation

https://www.fastify.io/docs/latest/Validation-and-Serialization/#serialization

nor any open tickets addressing this feature.

**Comment by mcollina** [comment]
https://www.fastify.io/docs/latest/Validation-and-Serialization/#schema-serializer should do the trick.

**Comment by Eomm** [comment]
This use case can be covered as follow-up PR after https://github.com/fastify/fastify/pull/2862 lands

See `serializerOpts`

> this.serializerCompiler = this.compilersFactory.buildSerializer(this.schemaBucket.getSchemas(), serverOption.serializerOpts)

I must push that PR forward in the next few days 👍🏽

**Comment by jsumners** [comment]
> [fastify.io/docs/latest/Validation-and-Serialization/#schema-serializer](https://www.fastify.io/docs/latest/Validation-and-Serialization/#schema-serializer) should do the trick.

Yes. Setting your own serializer should solve this as things are currently implemented.

**Comment by climba03003** [comment]
Resolved by #2862 

https://github.com/fastify/fastify/blob/main/docs/Server.md#serializeropts
